packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -29,17 +30,32 @@ contract Reserve is IReserve, Ownable, Initializable, UsingRegistry, ReentrancyG
   uint256 public constant TOBIN_TAX_DENOMINATOR = 1000;
   mapping(address => bool) public isSpender;
 
+  mapping(address => bool) public isCustodian;
+  address[] public _custodians;
+
+  bytes32[] public assetAllocationSymbols;
+  uint256[] public assetAllocationWeights;
+  uint256 public totalAssetAllocationWeight;
```

Comment:
Do we need to store this on-chain?

Reply:
Removed. Note that if you want to divide the assetAllocationWeights[] into e.g. 3rds, you need to handle rounding.